ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -167,24 +201,18 @@ public void execute(
           value
       );
 
-      producer.sendRecord(record, serviceContext, config.getProducerClientConfigProps());
-    } catch (final TopicAuthorizationException e) {
-      // TopicAuthorizationException does not give much detailed information about why it failed,
-      // except which topics are denied. Here we just add the ACL to make the error message
-      // consistent with other authorization error messages.
-      final Exception rootCause = new KsqlTopicAuthorizationException(
-          AclOperation.WRITE,
-          e.unauthorizedTopics()
-      );
-
-      throw new KsqlException("Failed to insert values into stream/table: "
-          + insertValues.getTarget().name(), rootCause);
-    } catch (final Exception e) {
-      throw new KsqlException("Failed to insert values into stream/table: "
-          + insertValues.getTarget().name(), e);
+      return record;
```

Comment:
nit: inline `record`:

```suggestion
      return new ProducerRecord<>(...);
```